staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -604,3 +604,32 @@ func poller(interval, timeout time.Duration) WaitFunc {
 		return ch
 	})
 }
+
+// ContextBoundExponentialBackoff works with a request context and a Backoff. It ensures that the retry wait never
+// exceeds the deadline specified by the request context.
+func ContextBoundExponentialBackoff(ctx context.Context, backoff Backoff, condition ConditionFunc) error {
```

Comment:
`ExponentialBackoffWithContext` is more in keeping with the rest of the names.

Reply:
This would have helped with [this recent search of mine](https://kubernetes.slack.com/archives/C0EG7JC6T/p1601514495044600) (from the "sig-api-machinery" Slack channel).